src/preconditions/ClientPermissions.ts

```diff
@@ -69,6 +71,8 @@ export class CorePrecondition extends AllFlowsPrecondition {
 	): AllFlowsPrecondition.AsyncResult {
 		const required = context.permissions ?? new PermissionsBitField();
 
+		if (!interaction.channel) return this.ok(); // When using user-installable apps channel will be always null
```

Comment:
Can't we still validate the permissions given `dmChannelPermissions`? Something like this:
```ts
let availablePermissions: PermissionsBitField;
if (interaction.channel) {
	const channel = await this.fetchChannelFromInteraction(interaction);
	availablePermissions = await this.getPermissionsForChannel(channel, interaction);
} else {
	availablePermissions = this.dmChannelPermissions;
}

return this.sharedRun(required, availablePermissions, '...');
```

And we could even refactor this into a private method:
```ts
private async getAvailablePermissions(interaction: BaseInteraction) {
	if (interaction.channel) {
		const channel = await this.fetchChannelFromInteraction(interaction);
		return this.getPermissionsForChannel(channel, interaction);
	}

	return this.dmChannelPermissions;
}
```

So then the methods are refactored to this:
```ts
const availablePermissions = await this.getAvailablePermissions(interaction);
return this.sharedRun(required, availablePermissions, '...');
```

At least this would help checking the permissions correctly, even if it's the base ones.

Reply:
If we don't have a channel, we should check appPermissions or dmChannelPermissions if that's also missing